src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,95 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.Arrays;
+import java.util.List;
+
+public class Parser {
+    public static Command parse(String text, State state) throws DukeException, DateTimeParseException {
+        if (text.contains("|")) {
+            throw new DukeException("No | pleaserin-o!");
+        }
+        List<String> parts = Arrays.asList(text.split(" "));
+        if (parts.isEmpty()) {
+            throw new DukeException("Invalid Commands");
+        }
+        String header = parts.get(0);
+        Command command;
+        switch(header) {
```

Comment:
Perhaps we can follow the Java coding standard provided on the CS2103/T website when it comes to switch statements? (The switch statement should have the following form: Note there is no indentation for case clauses.)
```` java
switch (condition) {
case ABC:
    statements;
    // Fallthrough
case DEF:
    statements;
    break;
case XYZ:
    statements;
    break;
default:
    statements;
    break;
}

Reply:
Thanks for the catch for this as well :)